src/eventer/eventer_SSL_fd_opset.c

```diff
@@ -1115,18 +1118,18 @@ void eventer_ssl_init() {
   SSL_library_init();
   OpenSSL_add_all_ciphers();
 
-  if (!dh512_file || strcmp(dh512_file, "")) {
+  if (!dh1024_file || strcmp(dh1024_file, "")) {
     e = eventer_alloc();
     e->mask = EVENTER_ASYNCH;
     e->callback = generate_dh_params;
-    e->closure = (void *)512;
+    e->closure = (void *)1024;
     eventer_add_asynch(NULL, e);
   }
-  if (!dh1024_file || strcmp(dh1024_file, "")) {
+  if (!dh2048_file || strcmp(dh2048_file, "")) {
```

Comment:
Since the DH param generation is non-blocking, and since you'll at least have 1024-bit params in < 1 second, I felt we could let the 2048-bit generation proceed.  It doesn't appear to hurt anything, and as soon as its done, new DHE connections will get the larger params.  We will be shipping a 2048-bit params file with our broker package, so we should have the larger params available immediately.